src/Core/Compiler/CompilerService.cs

```diff
@@ -28,98 +31,112 @@ namespace Microsoft.Quantum.IQSharp
 
-            return assembly;
+        public CompilerService()
+        {
+            this.Logger = new QSharpLogger(null);
+            this.CompilationManager = new CompilationUnitManager(ex => this.Logger?.Log(ex));
         }
 
         /// <summary>
         /// Compiles the given Q# code and returns the list of elements found in it.
         /// The compiler does this on a best effort, so it will return the elements even if the compilation fails.
         /// </summary>
-        public IEnumerable<QsCompiler.SyntaxTree.QsNamespaceElement> IdentifyElements(string source)
+        public IEnumerable<QsQualifiedName> IdentifyElements(string source)
         {
-            var ns = NonNullable<string>.New(Snippets.SNIPPETS_NAMESPACE);
-            var logger = new QSharpLogger(null);
-
-            var sources = new Dictionary<Uri, string>() { { new Uri($"file:///temp"), $"namespace {ns.Value} {{ {source} }}" } }.ToImmutableDictionary();
-            var references = QsReferences.Empty;
-
-            var loadOptions = new QsCompiler.CompilationLoader.Configuration(); // do not generate functor support
-            var loaded = new QsCompiler.CompilationLoader(_ => sources, _ => references, loadOptions, logger);
-            return loaded.VerifiedCompilation?.SyntaxTree[ns].Elements;
+            var nsName = NonNullable<string>.New(Snippets.SNIPPETS_NAMESPACE);
+            var content = $"namespace {nsName.Value} {{ {source} }}";
+            var fileManager = CompilationUnitManager.InitializeFileManager(Snippets.SNIPPET_FILE_URI, content);
+            var definedCallables = fileManager.GetCallableDeclarations();
+            var definedTypes = fileManager.GetTypeDeclarations();
+            return definedCallables.Concat(definedTypes).Select(decl => new QsQualifiedName(nsName, decl.Item1));
         }
 
-        /// <summary>
-        /// Builds the corresponding .net core assembly from the code in the given files.
-        /// </summary>
-        public AssemblyInfo BuildFiles(string[] files, CompilerMetadata metadatas, QSharpLogger logger, string dllName)
+        /// <summary> 
+        /// Compiles the given Q# code and returns the list of elements found in it. 
+        /// Removes all currently tracked source files in the CompilationManager and replaces them with the given ones.  
+        /// The compiler does this on a best effort, so it will return the elements even if the compilation fails. 
+        /// If the given references are not null, reloads the references loaded in by the CompilationManager  
+        /// if the keys of the given references differ from the currently loaded ones. 
+        /// Returns an enumerable of all namespaces, including the content from both source files and references.  
+        /// If generateFunctorSupport is set to true, replaces all auto-generation directives in the built syntax tree with the generated implementation. 
```

Comment:
Is `generateFunctorSupport` supposed to be a parameter to the method?

Reply:
Ah, sorry, it was but is no longer needed - will update the doc comment. Thanks!